convert2rhel/unit_tests/redhatrelease_test.py

```diff
@@ -96,36 +98,37 @@ def test_patch_yum_conf_existing_distroverpkg(version, monkeypatch):
         ("unknown", "anything", False),
     ),
 )
-def test_yum_is_modified(monkeypatch, pkg_type, subprocess_ret, expected_result):
+def test_pkg_manager_is_modified(monkeypatch, pkg_type, subprocess_ret, expected_result):
     monkeypatch.setattr(pkgmanager, "TYPE", value=pkg_type)
 
     run_subprocess = unit_tests.RunSubprocessMocked(return_string=subprocess_ret)
     monkeypatch.setattr(utils, "run_subprocess", value=run_subprocess)
+    pkg_manager_conf = redhatrelease.PkgManagerConf()
 
-    assert YumConf.is_modified() == expected_result
+    assert pkg_manager_conf.is_modified() == expected_result
 
 
 @pytest.mark.parametrize("modified", (True, False))
-def test_yum_patch(monkeypatch, modified, caplog):
+def test_pkg_manager_patch(monkeypatch, modified, caplog):
     is_modified = mock.Mock(return_value=modified)
-    monkeypatch.setattr(YumConf, "is_modified", value=is_modified)
+    monkeypatch.setattr(PkgManagerConf, "is_modified", value=is_modified)
     _comment_out_distroverpkg_tag = mock.Mock()
     monkeypatch.setattr(
-        YumConf,
+        PkgManagerConf,
         "_comment_out_distroverpkg_tag",
         value=_comment_out_distroverpkg_tag,
     )
-    _write_altered_yum_conf = mock.Mock()
-    monkeypatch.setattr(YumConf, "_write_altered_yum_conf", value=_write_altered_yum_conf)
+    _write_altered_pkg_manager_conf = mock.Mock()
```

Comment:
Change this mock to instead mock out `open()` somehow. Ideally mocking out the os part so it doesn't write into a file (faster that way)

Reply:
The test completes in .03 seconds so we likely won't save any noticable time by changing this

Reply:
@pr-watson point still stands. Right now it's mocking out functionality, meaning you're not testing anything. There is a function that is completely skipped now because of it. The idea behind mocking out the os part is to not create temporary files, as that will decrease the speed of the test. But the test should still make sure to execute through `_write_altered_pkg_manager_conf()`

Reply:
Wasn't able to avoid creating temp files but the test executes through _write_altered_pkg_manager_conf now without adding any noticable execution time